uiowa/uiowa
# Feature: horizontal nav dropdowns

Resolves #4823. 
Resolves #5061.

# To test
1. Setup:  
    a. `composer install`
    b. `blt frontend`
    c. Test on Tippie and Sandbox `blt ds --site=tippie.uiowa.edu` and `blt ds --site=sandbox.uiowa.edu --define drush.aliases.remote=sandbox.dev`
    d. `drush @tippie.local uli iowa-mba/curriculum-iowa-mba` and `drush @sandbox.local uli media`
    e. Go to the above pages to test out dropdowns. 
2. Tab through until you get to the menu (Shift+tab traverses backwards). 
    a.  If you hare focusing a menu dropdown indicator, space should toggle it open and closed.
    b.  If you hare focusing a menu dropdown indicator, enter should toggle it open and closed.
    c.  If a menu is open, the escape key should close it.
    d. All of the above should also work with nested menus.
3.  Test that the above works on mobile, and that the mobile version generally feels like it functions normally.
5. Test that alternative background colors still have reasonable button contrast.
6. Test that having that having the initial visibility level follow the active menu items, and both of the options therein, do not break functionality.
7. Test that stuffed menus work just fine. Fill them up as much as you can!
8. Examine `aria` 
9. test that <Nolink> links do not function oddly. 

<details>

# todo
- [x] Attach js based on `horizontal` selection possibly using https://git.drupalcode.org/project/layout_builder_styles/-/blob/8.x-1.x/src/EventSubscriber/BlockComponentRenderArraySubscriber.php#L96 as a way to do it. 
- [x] Add toggles to menu markup.
- [x] Add mobile menu toggle `<button id="main-menu-toggle" aria-label="Toggle main menu">Menu ☰</button>` (https://github.com/NickDJM/accessible-menu/blob/3.x/docs/basics/two-level-menu-with-toggle.md)
- [x] Deploy to dev and share with `NickDJM` for feedback on third-level
- [x] Test keyboard accesssibility

**Note:** I moved our initial prototype to Codepen for reference: https://codepen.io/bspeare/pen/ZEvymNR. 

## Testing/working on https://github.com/NickDJM/accessible-menu/pull/168
### Pull in feature branch
This needs to be run the first time you work on it or anytime you run `composer install`:
* `cd docroot/libraries/accessible-menu && git checkout top-link-disclosure-menu`
* If you make a change to the `src` JS files, do the following:
* `npm install`
* `npm run build`

Docs: https://accessible-menu.netlify.app/

</details>
